### PR TITLE
Refactor currency pkg

### PR DESCRIPTION
**Pull request by viettd**
- Remove redundant code
- Change `factor` to `float64` to avoid type casting
- Fix typo in `auto_version` script

### CODE REVIEW

File: currency/currency.go [hunk 1]
```diff
@@ -87,7 +87,7 @@ func ToMinorUnit(currencyCode string, amount float64) (minorUnitAmount uint64, e
 	switch {
 	case amount > 0:
 		factor := getCurrencyFactor(currencyCode)
-		minorUnitAmount = uint64(math.Round(amount * float64(factor)))
+		minorUnitAmount = uint64(math.Round(amount * factor))
 	case amount == 0:
 		minorUnitAmount = 0
 	default:
```
Comment (Pepito-Manaloto) [inline currency/currency.go:93]: do we also remove the `err` here same as `FromMinorUnit`?

File: currency/currency.go [hunk 2]
```diff
@@ -87,7 +87,7 @@ func ToMinorUnit(currencyCode string, amount float64) (minorUnitAmount uint64, e
 	switch {
 	case amount > 0:
 		factor := getCurrencyFactor(currencyCode)
-		minorUnitAmount = uint64(math.Round(amount * float64(factor)))
+		minorUnitAmount = uint64(math.Round(amount * factor))
 	case amount == 0:
```
Comment (Pepito-Manaloto) [inline currency/currency.go:91]: I think no need since the default value of uint is 0

Comment (viettd) [reply currency/currency.go:91]: `amount` is `float64` can be <= 0

File: currency/currency.go (hunk 1 above)
Comment (viettd) [reply currency/currency.go:93]: We cannot as `float64` can be <= 0

File: currency/currency.go (hunk 2 above)
Comment (Pepito-Manaloto) [reply currency/currency.go:91]: I mean the `case amount == 0`

Comment (viettd) [reply currency/currency.go:91]: Ah, I got what you meant now.
Let me make it shorter.

File: currency/currency.go [hunk 3]
```diff
@@ -99,19 +96,13 @@ func ToMinorUnit(currencyCode string, amount float64) (minorUnitAmount uint64, e
...
 func FromMinorUnit(currencyCode string, minorUnitAmount uint64) (amount float64, err error) {
 	if !IsISO4217(currencyCode) {
-		err = fmt.Errorf("%s is not a valid ISO 4217 currency code", currencyCode)
-		return
+		return 0, fmt.Errorf("%s is not a valid ISO 4217 currency code", currencyCode)
 	}
-	switch {
-	case minorUnitAmount > 0:
+
+	if minorUnitAmount > 0 {
```
Comment (martin-wego) [inline currency/currency.go:102]: Do we need to worry about minorUnitAmount < 0 ? 😮

Comment (Pepito-Manaloto) [reply currency/currency.go:102]: I think not possible since it the type is uint64

Comment (viettd) [reply currency/currency.go:102]: Yes, that why I removed it

**Review by Pepito-Manaloto — APPROVED** [review]

File: currency/currency.go (hunk 3 above)
Comment (martin-wego) [reply currency/currency.go:102]: Alright! I will resolve this and +1!

**Review by martin-wego — APPROVED** [review]

**Pull request merged**